packages/plugin-interactive-tools/sources/commands/search.tsx

```diff
@@ -0,0 +1,239 @@
+      const [page, setPage] = useState(0);
+      const [hits, setHits] = useState<Array<AlgoliaPackage>>([]);
+
+      const handleQueryOnChange = (newQuery: string) => {
+        // Ignore space and tab clicks
+        if (newQuery.match(/\t| /))
+          return;
+
+        setQuery(newQuery);
+      };
+
+      const fetchHits = async () => {
+        setPage(0);
+
+        const res = await search(query);
+
+        if (res.query === query) {
+          setHits(res.hits);
+        }
+      };
+
+      const fetchNextPageHits = async () => {
+        const res = await search(query, page + 1);
+
+        if (res.query === query && res.page - 1 === page) {
+          setPage(res.page);
+          setHits([...hits, ...res.hits]);
+        }
+      };
+
+      useEffect(() => {
+        if (!query) {
+          setHits([]);
+        } else {
+          fetchHits();
+        }
+      }, [query]);
+
+      // Typescript is having problems with
+      // recognizing InkTextInput as a valid
+      // JSX element for some reason...
+      const TextInput = InkTextInput as unknown as React.ComponentClass<InkTextInputProps>;
+
+      return <Box flexDirection={`column`}>
+        <Prompt />
+        <Box flexDirection={`row`}>
+          <Text bold>Search: </Text>
+          <Box width={41}>
+            <TextInput
+              value={query}
+              onChange={handleQueryOnChange}
+              placeholder={`i.e. babel, webpack, react...`}
+              showCursor={false}
+            />
+          </Box>
+          <SearchColumnNames />
+        </Box>
+        {hits.length ?
+          <ScrollableItems
+            radius={2}
+            loop={false}
+            children={hits.map(hit => <HitEntry key={hit.name} hit={hit} active={false} />)}
+            willReachEnd={fetchNextPageHits}
+          /> : <Color gray>Start typing...</Color>
+        }
+        <Box flexDirection={`row`}>
+          <Box width={49}>
+            <Text bold>Selected:</Text>
+          </Box>
+          <SelectedColumnNames />
+        </Box>
+        {selectedPackages.length ?
+          selectedPackages.map(
+            name => <SelectedEntry key={name} name={name} active={false}/>
+          ) : <Color gray>No selected packages...</Color>
+        }
+      </Box>;
+    };
+
+    const installRequests = await renderForm(SearchApp, {});
+    if (typeof installRequests === `undefined`)
+      return 1;
+
+    const dependencies = Array.from(installRequests.keys()).filter(request => installRequests.get(request) === `regular`);
+    const devDependencies = Array.from(installRequests.keys()).filter(request => installRequests.get(request) === `dev`);
+    const peerDependencies = Array.from(installRequests.keys()).filter(request => installRequests.get(request) === `peer`);
+
+    await this.cli.run([`add`, ...dependencies]);
+    await this.cli.run([`add`, `--dev`, ...devDependencies]);
+    await this.cli.run([`add`, `--peer`, ...peerDependencies]);
```

Comment:
This won't work if those arrays are empty. Also, it might be a better idea to directly update the descriptors inside the workspace manifest and run an install, rather than sending them through `yarn add`. It would be more consistent with how `yarn upgrade-interactive` (refer to it for the implementation) works and I think it would also be faster (since we already have the versions, we can just make descriptors rather than sending them through the suggestion engine).

BTW, just to make sure you understand, descriptors consist of an optional scope, a name, and a range (which is the version of the package in this case).

Reply:
I haven't looked at the PR yet, but fwiw I'm the one who suggested to leverage `yarn add`, to make the implementation easier.

Reply:
> This won't work if those arrays are empty.

Stupid mistake 🤦🏽‍♂️

> Other part.

Yeah, I was thinking about that. I looked at the `add` command implementation and saw that there is quite a bit on functionality there that I thought would not make sense to re-implement. I will look into it again though, the version part makes sense and there probably is way to make it faster but still slick.

I'll get back to you in this comment.

Reply:
@arcanis Didn't know you suggested this, I probably missed the discussion. 😅

🤔 Maybe sending them through `yarn add` is actually the right way to do this (e. g. for peers we shouldn't use the version, but `*` instead), because it would save that complexity.
Let's just hope that nobody publishes a new version in the millisecond between selecting the packages and running `yarn add`.

Reply:
Another thing to note is that 3 reports will be outputted if you add at least one of each kind of dependency, and that might be a bit excessive, at least for repos like ours, where each install report contains 3 warnings and 20 "build scripts disabled" messages.

Reply:
> Another thing to note is that 3 reports will be outputted if you add at least one of each kind of dependency, and that might be a bit excessive, at least for repos like ours, where each install report contains 3 warnings and 20 "build scripts disabled" messages.

I noticed that yeah, not sure how to fix it though since there is no way to add different types of dependencies with a single `add` command.

Reply:
> > Let's just hope that nobody publishes a new version in the millisecond between selecting the packages and running yarn add. (So that you expect to install react 16, but suddenly react 17 is published)

After all, there's the same risk of race conditions between the time you make the search on the Yarn website and you run `yarn add` 😄

> 20 "build scripts disabled" messages.

Those really should be silenced after the first time. Whether to run them or not is handled within the linkers, which aren't aware of the build state, but it should instead be part of the project builder. This way we would be able to store that we didn't run them purposely.

Reply:
> Those really should be silenced after the first time. Whether to run them or not is handled within the linkers, which aren't aware of the build state, but it should instead be part of the project builder. This way we would be able to store that we didn't run them purposely.

I'm not sure I'm following, so using the `add` command should be the right way to go?